libres/tests/analysis/test_update.cpp

```diff
@@ -0,0 +1,120 @@
+#include <string>
+#include <unordered_map>
+#include <vector>
+#include <cmath>
+
+#include <catch2/catch.hpp>
+
+#include <ert/util/rng.h>
+#include <ert/enkf/enkf_util.hpp>
+
+#include <ert/analysis/update.hpp>
+
+namespace analysis {
+void run_analysis_update_without_rowscaling(
+    analysis_module_type *module, const bool_vector_type *ens_mask,
+    const meas_data_type *forecast, obs_data_type *obs_data,
+    rng_type *shared_rng, matrix_type *E,
+    std::unordered_map<std::string, matrix_type *> parameters);
+}
+
+const double a_true = 5.0;
+const double b_true = 0.5;
+
+struct model {
+    double a;
+    double b;
+
+    model(double a, double b) : a(a), b(b) {}
+
+    model(rng_type *rng) {
+        this->a = enkf_util_rand_normal(a_true, 0.5, rng);
+        this->b = enkf_util_rand_normal(b_true, 0.5, rng);
+    }
+
+    double eval(double x) const { return this->a * x + this->b; }
+};
+
+SCENARIO("Running analysis update without row scaling on linear model",
+         "[analysis]") {
+    GIVEN("Increasing ensemble size") {
+        int ens_size = GENERATE(10, 100, 1000);
+        int obs_size = 2;
+
+        std::vector<model> ens;
+
+        analysis_module_type *ies_module = analysis_module_alloc("IES_ENKF");
+
+        bool_vector_type *ens_mask = bool_vector_alloc(ens_size, true);
+        obs_data_type *obs_data = obs_data_alloc(1.0);
+        meas_data_type *meas_data = meas_data_alloc(ens_mask);
+        rng_type *rng = rng_alloc(MZRAN, INIT_DEFAULT);
+
+        model true_model{a_true, b_true};
+
+        for (int iens = 0; iens < ens_size; iens++)
+            ens.push_back(model(rng));
+
+        const char *obs_key = "OBS1";
+        meas_block_type *mb =
+            meas_data_add_block(meas_data, obs_key, 1, obs_size);
+        obs_block_type *ob =
+            obs_data_add_block(obs_data, obs_key, obs_size, nullptr, false);
+
+        for (int iobs = 0; iobs < obs_size; iobs++) {
+            double x = iobs;
+            for (int iens = 0; iens < ens_size; iens++) {
+                const auto &m = ens[iens];
+                meas_block_iset(mb, iens, iobs, m.eval(x));
+            }
+
+            obs_block_iset(ob, iobs, true_model.eval(x), 0.25);
+        }
+
+        matrix_type *E = obs_data_allocE(obs_data, rng, ens_size);
+
+        int nparam = 2;
+        matrix_type *A = matrix_alloc(nparam, ens_size); // prior
+        for (int iens = 0; iens < ens_size; iens++) {
+            matrix_iset(A, 0, iens, enkf_util_rand_uniform(0, 10, rng));
```

Comment:
This is wrong. In your current setup there is no coupling between the model parameters and used to "simulate" with 
```C++
meas_block_iset(mb, iens, iobs, m.eval(x));
```
and the parameters which are passed to the update routine. Instead of sampling random values for the `A` matrix the initialization of the `A` matrix should be:
```C++
for (int iens = 0; iens < ens_size; iens++) {
     const auto& model = ens[iens];
     matrix_iset(A, 0, iens, model.a);
     matrix_iset(A, 1, iens, model.b);
}
```
in terms of the notation used in ert code the `struct model { a,b }` maps to a `enkf_node` instance, and the loop above corresponds to `serialize` step.

Reply:
Initially I had them set as follows:

```python
            matrix_iset(A, 0, iens, enkf_util_rand_normal(a_true, 0.5, rng));
            matrix_iset(A, 1, iens, enkf_util_rand_normal(b_true, 0.5, rng));
```

My thinking was that `A` contains the priors for the two parameters `a` and `b`, where I assumed they were normally distributed around `a_true` and `b_true`.

Then I wanted to try and mimic the poly_example and set the priors using a uniform distribution.
This is a misunderstanding on my part.

If I understand you correctly, the parameter matrix `A` should contain the same parameters that were used to make the forecast in the `meas_block` called `mb`?

Reply:
> My thinking was that A contains the priors for the two parameters a and b

That is correct - but the same values must be used to run the forward model and generate the simulated values.If I 

>If I understand you correctly, the parameter matrix A should contain the same parameters that were used to make the forecast in the meas_block called mb?

YES!